File: src/main/java/com/webbasedtourguide/enums/UserType.java
package com.webbasedtourguide.enums;

import com.webbasedtourguide.repositories.TouristRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.repository.CrudRepository;

public enum UserType {
    TOURIST,
    TOURGUIDE,
    AGENCYSTAFF,
    TOURMANAGER
}

File: frontend/src/enums/UserType.ts
export const UserType = {
    Tourist: "TOURIST",
    TourGuide: "TOURGUIDE",
    AgencyStaff: "AGENCYSTAFF",
    TourManager: "TOURMANAGER",
} as const;

export const ValidUserTypes = {
    Tourist: "TOURIST",
    TourGuide: "TOURGUIDE",
    AgencyStaff: "AGENCYSTAFF",
    TourManager: "TOURMANAGER",
} as const;

export type UserType = (typeof UserType)[keyof typeof UserType];
export type ValidUserTypes = (typeof ValidUserTypes)[keyof typeof ValidUserTypes];